benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,647 @@
+    sampler.flush()
+    end_time = perf_counter()
+    print('flushed all batches')
+    return (end_time - start_time)
+
+
+def assign_offsets_pyg(node_counts: Dict[str, int], replication_factor:int=1):
+    # cuGraph-PyG assigns offsets based on lexicographic order
+    node_offsets = {}
+    node_offsets_replicated = {}
+    count = 0
+    count_replicated = 0
+    for node_type in sorted(node_counts.keys()):
+        node_offsets[node_type] = count
+        node_offsets_replicated[node_type] = count_replicated
+
+        count += node_counts[node_type]
+        count_replicated += node_counts[node_type] * replication_factor
+    
+    return node_offsets, node_offsets_replicated, count_replicated
+
+def generate_rmat_dataset(dataset, seed=62, labeled_percentage=0.01, num_labels=256, reverse_edges=False):
+    """
+    Generates an rmat dataset.  Currently does not support heterogeneous datasets.
+
+    Parameters
+    ----------
+    dataset: The specifier of the rmat dataset (i.e. rmat_20_16)
+    seed: The seed to use for random number generation
+    num_labels: The number of classes for the labeled nodes
+    reverse_edges: Whether to reverse the edges in the edgelist (should be True for DGL, False, for PyG)
+    """
+
+    dataset = dataset.split('_')
+    scale = int(dataset[1])
+    edgefactor = int(dataset[2])
+
+    dask_edgelist_df = generate_edgelist_rmat(
+        scale=scale, edgefactor=edgefactor, seed=seed, unweighted=True, mg=True,
+    )
+    dask_edgelist_df = dask_edgelist_df.astype("int64")
+    dask_edgelist_df = dask_edgelist_df.reset_index(drop=True)
+
+
+    dask_edgelist_df = renumber_ddf(dask_edgelist_df).persist()
+    dask_edgelist_df = symmetrize_ddf(dask_edgelist_df).persist()
+    dask_edgelist_df['etp'] = cupy.int32(0) # doesn't matter what the value is, really
+    
+    # generator = np.random.default_rng(seed=seed)
+    num_labeled_nodes = int(2**(scale+1) * labeled_percentage)
+    label_df = pd.DataFrame({
+        'node': np.arange(num_labeled_nodes),
+        # 'label': generator.integers(0, num_labels - 1, num_labeled_nodes).astype('float32')
+    })
+    
+    n_workers = len(default_client().scheduler_info()['workers'])
+    dask_label_df = ddf.from_pandas(label_df, npartitions=n_workers*2)
+    del label_df
+    gc.collect()
+
+    dask_label_df = dask_cudf.from_dask_dataframe(dask_label_df)
+
+    node_offsets = {'paper': 0}
+    edge_offsets = {('paper','cites','paper'):0}
+    total_num_nodes = int(dask_cudf.concat([dask_edgelist_df.src, dask_edgelist_df.dst]).nunique().compute())
+
+    if reverse_edges:
+        dask_edgelist_df = dask_edgelist_df.rename(columns={'src':'dst', 'dst':'src'})
+
+    return dask_edgelist_df, dask_label_df, node_offsets, edge_offsets, total_num_nodes
+
+
+def load_disk_dataset(dataset, dataset_dir='.', reverse_edges=True, replication_factor=1):
+    path = os.path.join(dataset_dir, dataset)
+    parquet_path = os.path.join(path, 'parquet')
```

Comment:
If you're interested in cleaning up all the `os.path.join(...)` calls, you could make these `pathlib.Path` objects then use the slash operator
```suggestion
    from pathlib import Path
    path = Path(dataset_dir) / dataset
    parquet_path = path / 'parquet'
```